r/content/specify_data_types_and_schemas.Rmd

```diff
@@ -0,0 +1,279 @@
+})
+```
+
+## Specify data types when creating an Arrow table from an R object
+
+You want to manually specify Arrow data types when converting an object from a data frame to an Arrow object.
+
+### Solution
+
+```{r, use_schema}
+# Set up a tibble to use in this example
+oscars <- tibble::tibble(
+  actor = c("Katharine Hepburn", "Meryl Streep", "Jack Nicholson"),
+  num_awards = c(4, 3, 3)
+)
+
+# Set up schema with "num_awards" as integer
+oscars_schema <- schema(actor = string(), num_awards = int16())
+
+# create arrow Table containing data and schema
+oscars_data_arrow <- Table$create(oscars, schema = oscars_schema)
+
+oscars_data_arrow
+```
+```{r, test_use_schema, opts.label = "test"}
+test_that("use_schema works as expected", {
+  expect_s3_class(oscars_data_arrow, "Table")
+  expect_equal(
+    oscars_data_arrow$schema,
+    oscars_schema
+  )
+})
+```
+
+## Specify data types when reading in files
+
+You want to manually specify Arrow data types when reading in files.
+
+### Solution
+
+```{r, use_schema_dataset}
+# Set up a tibble to use in this example
+oscars <- tibble::tibble(
+  actor = c("Katharine Hepburn", "Meryl Streep", "Jack Nicholson"),
+  num_awards = c(4, 3, 3)
+)
+
+# write dataset to disk
+write_dataset(oscars, path = "oscars_data")
+
+# Set up schema with "num_awards" as integer
+oscars_schema <- schema(actor = string(), num_awards = int16())
+
+# read the dataset in, using the schema instead of inferring the type automatically
+oscars_dataset_arrow <- open_dataset("oscars_data", schema = oscars_schema)
+
+oscars_dataset_arrow
+```
+```{r, test_use_schema_dataset, opts.label = "test"}
+test_that("use_schema_dataset works as expected", {
+  expect_s3_class(oscars_dataset_arrow, "Dataset")
+  expect_equal(oscars_dataset_arrow$schema,
+    oscars_schema
+  )
+})
+```
+```{r, include=FALSE}
+unlink("oscars_data", recursive = TRUE)
+```
+
+## Combine and unify schemas
+
+You have a dataset split across multiple sources for which you have separate 
+schemas that you want to combine.
+
+### Solution
+
+You can use `unify_schemas()` to combine multiple schemas into a single schemas.
+
+```{r, combine_schemas}
+# create first schema to combine
+country_code_schema <- schema(country = utf8(), code = utf8())
+
+# create second schema to combine
+country_location_schema <- schema(longitude = float32(), latitude = float32())
+
+# combine schemas
+combined_schemas <- unify_schemas(country_code_schema, country_location_schema)
+combined_schemas
+```
+
+```{r, test_combine_schemas, opts.label = "test"}
+test_that("combine_schemas works as expected", {
+  expect_s3_class(combined_schemas, "Schema")
+  expect_equal(combined_schemas,
+    schema(country = utf8(), code = utf8(), longitude = float32(), latitude = float32())
+  )
+})
+```
+
+
+### Discussion
+
+If you have two schemas containing fields with the same names and types, you can
+ combine them into a single field.
```

Comment:
```suggestion
If you have two different schemas with some overlapping fields which have the same names and types, you can
 combine them into a single schema.
```

Nit: I think I get what you are after but off the cuff this wasn't clear to me.  I don't feel too strongly about it though.